BeeSwift/Components/GoalImageView.swift

```diff
@@ -68,78 +69,40 @@ class GoalImageView : UIView {
-        // Invalidate the download token, meaning that any queued download callbacks
-        // will no-op. This avoids race conditions with downloads finishing out of order.
-        let newDownloadToken = UUID()
-        self.currentDownloadToken = newDownloadToken
-
-        if let downloadReceipt = inProgressDownload {
-            GoalImageView.downloader.cancelRequest(with: downloadReceipt)
-            inProgressDownload = nil
-        }
-
         //  - Deadbeat: Placeholder, no animation
-        if ServiceLocator.currentUserManager.isDeadbeat() {
+        guard !ServiceLocator.currentUserManager.isDeadbeat() else {
             clearGoalGraph()
             return
         }
 
         //  No Goal: Placeholder, no animation
-        guard let goal = self.goal else {
+        guard let goal else {
             clearGoalGraph()
             return
         }
 
+        // Load the appropriate image for the goal
+        let urlString = isThumbnail ? goal.cacheBustingThumbUrl : goal.cacheBustingGraphUrl
+        let request = URLRequest(url: URL(string: urlString)!)
+        
         // When queued, we should show a loading indicator over any existing graph,
         // but not over the placeholder image.
-        if goal.queued {
-            beeLemniscateView.isHidden = !currentlyShowingGraph
-        }
-
-        // Load the approppriate iamge for the goal
-        let urlString = if isThumbnail {
-            goal.cacheBustingThumbUrl
-        } else {
-            goal.cacheBustingGraphUrl
-        }
-        let request = URLRequest(url: URL(string: urlString)!)
-
-        // Explicitly check the cache to see if the image is already present, and if so set it directly
-        // This avoids flicker when showing images from the cache, which will otherwise briefly show
-        // the placeholder while waiting for the async download callback
-        if let image = GoalImageView.downloader.imageCache?.image(for: request, withIdentifier: nil) {
-            showGraphImage(image: image)
-            return
-        }
-
-        // Download the image and show it once downloaded
-        inProgressDownload = GoalImageView.downloader.download(request, completion: { response in
-            if newDownloadToken != self.currentDownloadToken {
-                // Another refresh has happend since we were enqueued. Skip performing any updates
-                return
-            }
-
-            switch(response.result) {
-            case .success(let image):
-                // Image downloaded. Show it, and have loading indicator match queued state
-                self.showGraphImage(image: image)
-                break;
-            case .failure(let error):
-                self.logger.error("Error downloading goal graph: \(error)")
-                self.clearGoalGraph()
-            }
-        })
+        beeLemniscateView.isHidden = !currentlyShowingGraph
+        
+        imageView.af.setImage(withURLRequest: request,
```

Comment:
And here where it is being used in the Today View extension: https://github.com/krugerk/BeeSwift/blob/0456597a73674957ebd7734a7518cf6625f57d25/BeeSwiftToday/TodayTableViewCell.swift#L190